apps/wing/src/analytics/collect.ts

```diff
@@ -0,0 +1,93 @@
+import { Command } from "commander";
+import { loadAnalyticsConfig, storeAnalyticEvent } from "./storage";
+import { AnalyticEvent } from "./event";
+import { OSCollector } from "./collectors/os-collector";
+import { NodeCollector } from "./collectors/node-collector";
+import { CLICollector } from "./collectors/cli-collector";
+import { CICollector } from "./collectors/ci-collector";
+import { PACKAGE_VERSION } from "../cli";
+
+
+/**
+ * Collects analytics for a given command, stores it for later export
+ * 
+ * @param cmd The commander command to collect analytics for
+ * @returns string the file path of the stored analytic
+ */
+export async function collectCommandAnalytics(cmd: Command): Promise<string | undefined> {
+  const osCollector = new OSCollector();
+  const nodeCollector = new NodeCollector();
+  const ciCollector = new CICollector();
+  const cliCollector = new CLICollector(cmd);
+
+  let event: AnalyticEvent = {
+    event: `wing cli:${cmd.name()}`,
```

Comment:
Some naming convention change, it is <source>_<object>_<action>
in this case:
-  source is `cli`
- object is `--target` 
- action is `test/compile/etc`
 
if object (target) is empty (like in `wing docs` it should be "nil"), unless there is a default (then it should be the default) 
Also, lets remove anything in the object/action that is not normal chars [a-zA-Z] `str.replace(/[^a-zA-Z]/g, '');`

meaning it should be:
- `name: cli_sim_test`
- `name: cli_tfaws_compile`
- `name: cli_sim_compile`
- `name: cli_tfaws_compile`

Reply:
sorry not sure if I get this completely. In the above are `cli_sim_test`, `cli_tfaws_compile`, etc.. values in `event`? Also does that mean that running docs would produce `cli_nil_docs` or `cli_docs` or `cli__docs`

Reply:
Let's go with `cli__docs`, I think it is the best approach

Reply:
alright heres what it looks like
<img width="839" alt="image" src="https://github.com/winglang/wing/assets/45375125/42a362cc-4f90-4c61-bdec-310b4cdbeac3">